lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -50,21 +49,27 @@ sub latest_build {
 
     $attrs{join}     = 'settings';
     $attrs{rows}     = 1;
-    $attrs{order_by} = {-desc => 'me.id'};              # More reliable for tests than t_created
-    $attrs{columns}  = [{value => 'settings.value'}];
-    push(@conds, {'settings.key' => {'=' => 'BUILD'}});
+    $attrs{order_by} = {-desc => 'me.id'};    # More reliable for tests than t_created
+    $attrs{columns}  = qw/BUILD/;
 
     while (my ($k, $v) = each %args) {
-        my $subquery = $schema->resultset("JobSettings")->search(
-            {
-                key   => uc($k),
-                value => $v
-            });
-        push(@conds, {'me.id' => {-in => $subquery->get_column('job_id')->as_query}});
+
+        if (grep { $k eq $_ } qw/distri version flavor machine arch build test/) {
```

Comment:
no, this is query. and you can query for ISO and DESKTOP - different table for those